jimp.d.ts

```diff
@@ -30,285 +166,413 @@ declare namespace Jimp {
             cb?: Jimp.ImageCallback
         );
 
-        bitmap: Bitmap;
-
-        clone(cb?: Jimp.ImageCallback): Jimp;
-        quality(n: number, cb?: Jimp.ImageCallback): this;
-        deflateLevel(l: number, cb?: Jimp.ImageCallback): this;
-        deflateStrategy(s: number, cb?: Jimp.ImageCallback): this;
-        filterType(f: number, cb?: Jimp.ImageCallback): this;
-
-        rgba(bool: boolean, cb?: Jimp.ImageCallback): this;
-        background(hex: number, cb?: Jimp.ImageCallback): this;
-        scan(
-            x: number,
-            y: number,
-            w: number,
-            h: number,
-            f: (x: number, y: number, idx: number) => any,
-            cb?: Jimp.ImageCallback
-        ): this;
-        getWidth(): number;
+        // Methods
         getHeight(): number;
+        getWidth(): number;
+        inspect(): string;
+        toString(): string;
         getMIME(): string;
         getExtension(): string;
-        getPixelIndex(
+        write(path: string, cb?: Jimp.ImageCallback): this;
+        writeAsync(path: string): Promise<Jimp>;
+        deflateLevel(l: number): this;
+        deflateLevel<T>(l: number, cb: Jimp.ImageCallback<T>): T;
+        deflateStrategy(s: number): this;
+        deflateStrategy<T>(s: number, cb: Jimp.ImageCallback<T>): T;
+        filterType(f: number): this;
+        filterType<T>(f: number, cb: Jimp.ImageCallback<T>): T;
+        rgba(bool: boolean): this;
+        rgba<T>(bool: boolean, cb: Jimp.ImageCallback<T>): T;
+        quality(n: number): this;
+        quality<T>(n: number, cb: Jimp.ImageCallback<T>): T;
+        getBase64(mime: string, cb?: GenericCallback<string, any, this>): this;

```

Comment:
Why is `GenericCallback` used in some places? They seem to have the same style callback as the rest

Reply:
JimpCallback is just an alias for a GenericCallback with a Jimp instance as the passed value. I could've used `GenericCallback<Jimp, T, Jimp>` everywhere, but it was common enough to make it a separate type